Src/BootCamp.Chapter/Program.cs

```diff
@@ -1,12 +1,264 @@
+            // if W, A, S, D is pressed will print corresponding character on screen
+            // until Escape is pressed then will return the user to the main menu
+            while (true)
+            {
+                ConsoleKey pressedKey = Console.ReadKey(true).Key;
+
+                if (pressedKey == ConsoleKey.Escape)
+                {
+                    MainMenu();
+                }
+
+                if (pressedKey == ConsoleKey.W)
+                {
+                    Console.Write(arrowUp);
+                }
+                if (pressedKey == ConsoleKey.S)
+                {
+                    Console.Write(arrowDown);
+                }
+                if (pressedKey == ConsoleKey.A)
+                {
+                    Console.Write(arrowLeft);
+                }
+                if (pressedKey == ConsoleKey.D)
+                {
+                    Console.Write(arrowRight);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates sub-menu for Decimal to Binary conversion
+        /// </summary>
+        private static void ConversionMenuOption()
+        {
+            Console.CursorVisible = false;
+
+            Console.Clear();
+            Console.WriteLine("Conversions");
+            Console.WriteLine("  (1) Conversion =>");
+            Console.WriteLine("  (2) Play with WASD arrows.");
+            Console.WriteLine("(Esc) Back to Main Menu!");
+
+            while (true)
+            {
+                ConsoleKey pressedKey = Console.ReadKey(true).Key;
+
+                if (pressedKey == ConsoleKey.Escape)
+                {
+                    MainMenu();
+                }
+                if (pressedKey == ConsoleKey.D1 || pressedKey == ConsoleKey.NumPad1)
+                {
+                    DecToBinOption();
+                    break;
+                }
+                if (pressedKey == ConsoleKey.D2 || pressedKey == ConsoleKey.NumPad2)
+                {
+                    BinToDecOption();
+                    break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates sub-menu for Decimal to Binary conversion
+        /// </summary>
+        private static void DecToBinOption()
+        {
+            Console.CursorVisible = true;
+            Console.Write("Enter integer number: ");
+
+            while (true)
+            {
+                string input = Console.ReadLine();
+
+                if (string.IsNullOrEmpty(input))
+                {
+                    Console.WriteLine("Null or Empty string not allowed!");
+                    Thread.Sleep(displayTimeout);
+                    ConversionMenuOption();
+                }
+
+                if (IsNumber(input))
+                {
+                    int dec = StringToInt(input);
+                    Console.WriteLine($"Converted from decimal to binary result: {ConvertDecToBin(dec)}");
+                    Thread.Sleep(displayTimeout);
```

Comment:
These 2 last lines are repeated in both if and else. They could be moved to the outside of ifs